lib/rollout.rb

```diff
@@ -159,6 +166,33 @@ def define_group(group, &block)
     @groups[group.to_sym] = block
   end
 
+  def define_id_collection(name, ids = [])
+    @storage.set("collection:#{name}", ids.join(","))
+    @storage.set(collection_key, (collections.keys | [name.to_sym]).join(","))
+  end
+
+  def add_collection_to_feature(collection, feature)
+    with_feature(feature) do |feature|
+      @storage.set(collection_feature_key(feature.name), (feature_collections(feature.name).keys | [ collection.to_sym]).join(","))
+    end
+  end
+
+  def collections
+    Hash[*retrieve_from_storage(collection_key).map{ |collect| [ collect.to_sym, @storage.get("collection:#{collect}") ] }.flatten]
```

Comment:
An example:
this method will retrieve a hash with all collections with key being the collection as key, and the array of user_ids as value. So:
Hash[\***retrieve_from_storage(collection_key)**.map{ |collect| [ collect.to_sym, @storage.get(collection_key(collect)) ] }.flatten]
This will return all collections from `Redis`, like: `=> ["beta_testers", "my_collection"]`.
After that, I'll map this to retrieve all users_ids from each collection:
Hash[\* ["beta_testers", "my_collection"].**map{ |collect| [ collect.to_sym, @storage.get(collection_key(collect)) ] }**.flatten].
The map will return an array, of arrays, like:
`[[:beta_testers, "101,103"], [:my_collection, "101,102"]]`.
After that, a `flatten` will be applied on it:
[[:beta_testers, "101,103"], [:my_collection, "101,102"]].**flatten!**
resulting on something like:
`[:beta_testers, "101,103", :my_collection, "101,102"]`
And the Hash[ `array` ] will transform it on a hashmap.
the final result will be:
`{:beta_testers=>"101,103", :my_collection=>"101,102"}`
So, with this, I can access the `key` of the collection and check if a user_id belongs on the collection.

@GonchuB, what you think? :smile: